src/com/sap/piper/JenkinsUtils.groovy

```diff
@@ -108,3 +109,21 @@ String getIssueCommentTriggerAction() {
 def getJobStartedByUserId() {
     return getRawBuild().getCause(hudson.model.Cause.UserIdCause.class)?.getUserId()
 }
+
+@NonCPS
+def getLibrariesInfo() {
+    def libraries = []
+    def build = getRawBuild()
+    def libs = build.getAction(LibrariesAction.class).getLibraries()
+
+    for (def i = 0; i < libs.size(); i++) {
+        Map lib = [:]
+
+        lib['name'] = libs[i].name
+        lib['version'] = libs[i].version
```

Comment:
Would be cool to get - in addition - the git sha for the version (which will be typically a tag or `master). With that we can compare to the version of the go binary. In case we compare for tags we are on the save side since tags are expected to be stable. But for the `master` branch it is not sufficient to check for the branch name since the branch moves and the sha is not stable.

In the "manage jenkins" view we have for each library something like "Currently maps to revision: xxxxx". Hence that is in general possible.

Reply:
I think having 80% covered is not sufficient. I also think for most of the cases we won't have really incompatibilities. But nevertheless we will have some. And we have to support this. For such support cases it simplifies our job a a lot if there is a clear warning in the log indicating that there is a mismatch (maybe even raising an exception in that case might be reasonable). Scenario: we merge something to `master`. Some seconds later somewhere on this planet on customer side having `master` configured a build is triggered downloading the - at that point in time - current version of the piper binary. Some seconds later our build process updates the piper binary to the now up to date version. But on customer side the old version has already been materialized, --> 🔥 

I'm  not against the current approach, but we should urgently find a way to get the current git sha for the piper lib so that we can check on the level without doubt if the piper binary matches to the groovy part.

Reply:
What do we do with the info when we see that it does not match?
As you say, in most cases this is no issue ...

Reply:
> What do we do with the info when we see that it does not match?

We should have at least a warning in the log. This simplifies troubleshooting. Having clear error messages or hints in the log makes a big difference. With a good log a user might be able to deal with the issue directly, without we might have a bigger time gap where the issue bubbles up and down before someone finds the reason. A warning is the minimum IMO. If there should be a build failure in that case - good question. There are pros and cons. A pro is e.g. there might be cases where the build does not directly fail but produces some unexpected binaries or other side effects which hits much later. I know that is not very likely, but it is possible. How we have have to balance that. In such cases where there are valid pros and cons it is not possible to take a good decission for all use-cases. In such a case it might be an option to make that configurable ( ... knowing that not all colleagues are happy with a huge set of config options).